// containers/ProductDetailContainer.js
import React, { Component } from 'react'
import { connect } from 'react-redux'

import { getProductById } from 'selectors'
import { fetchProductById } from 'actions'


class ProductDetailContainer extends Component {
    componentDidMount () {
        this.props.fetchProductById(this.props.id)
    }

    renderContent () {
        const {product} = this.props
        return (
            <div>
                <p>Zurück</p>
                <div>
                    <h1>{product.name}</h1>
                    <p>{product.price}</p>
                    <p>{product.description}</p>
                    <p>
                        ### Sauce ###<br/>
                        ### Anzahl ###
                    </p>
                    <p>Add to Cart</p>
                </div>
                <div>
                    <img
                        className="product-detail-image"
                        src={product.image}
                        alt={product.name}
                    />
                </div>
            </div>
        )
    }

    render () {
        const {product} = this.props
        // console.log('detailseite product', product);
        return (
            <div>
                {product && this.renderContent()}
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        urlId: state.id,
        product: getProductById(state, state.productPage.id)
    }
}

const mapDispatchToProps = {
    fetchProductById
}

export default connect(mapStateToProps, mapDispatchToProps)(ProductDetailContainer)
